### PR TITLE
Add the auto_update option to the npm_package resource.

### PR DESCRIPTION
**Pull request by MarkGibbons**
Allows stopping the default behavior of doing an update if possible.

### Description

[Describe what this change achieves]

### Issues Resolved

Replaces PR #159

### Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # NodeJS Cookbook Changelog
 
+## unreleased
```
Comment (ramereth) [inline CHANGELOG.md:3]:
```suggestion
## Unreleased
```

**Review by ramereth — COMMENTED** [review]
LGTM otherwise.

**Review by ramereth — APPROVED** [review]

**Comment by ramereth** [comment]
@MarkGibbons do you want this released now or wait for the other PRs to be merged?

**Comment by MarkGibbons** [comment]
I'd like to see 222 and 226 released together.  The other changes that depend on 222 will take me a week or so.  We can release now or then.  Thank you.

**Comment by ramereth** [comment]
Let's just wait until you get it all done then.

**Comment by wjordan** [comment]
Note: I believe this PR not only added a configuration option but also changed the default behavior, starting in v7.3.0.

The existing `not_if { package_installed? }` clause meant that the default was to _not_ run `npm install` when the package had already been installed. With this PR, the default changes to auto-upgrade packages while allowing the previous behavior to be configured by setting the newly-added `auto_update false` option.

This affected my application following an upgrade of the `nodejs` cookbook, existing installed npm packages were suddenly auto-updated to their latest versions. Perhaps it's worth calling out this change in the documentation more clearly?

**Pull request merged**